### PR TITLE
Proposed changes for PR 8408

### PR DESCRIPTION
**Pull request by peterfriese**
Hi Flo,

I went ahead and implemented some of the changes I suggested in my review to make it easier to see what I mean. Feel free to use or discard at your own discretion.

Cheers,
Peter

### CODE REVIEW

File: Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift [hunk 1]
```diff
@@ -0,0 +1,187 @@
+/*
```
Comment (chFlorian) [inline Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift:1]: Could you pull the current version of my branch (so these changes get removed)?
Then we could merge this example project into the main PR.

File: Firestore/Swift/Source/PropertyWrapper/QueryPredicate.swift [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+
+/// Query predicates that can be used to filter results fetched by `FirestoreQuery`.
+///
+/// Construct predicates using one of the following ways:
+///
+///     let onlyFavourites: QueryPredicate = .whereField("isFavourite", isEqualTo: true)
```
Comment (chFlorian) [inline Firestore/Swift/Source/PropertyWrapper/QueryPredicate.swift:23]: Do you want to leave in all 3 ways in the final rlease or is this as a preparation for the api review feedback?

File: Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+/// This property wrapper does not support updating the `wrappedValue`, i.e.
+/// you need to use Firestore's other APIs to add, delete, or modify documents.
+@available(iOS 14.0, *)
+@available(tvOS, unavailable)
+@propertyWrapper
+public struct FirestoreQuery<T: Decodable>: DynamicProperty {
+  @StateObject private var firestoreQueryObservable: FirestoreQueryObservable<T>
+  
+  /// The query's configurable properties.
+  public struct Configuration {
```
Comment (chFlorian) [inline Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift:63]: Is this part of your style guidelines to place the struct here? Personally I would keep it in an extension for better separation.

File: Example/FirestoreSample/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Firestore Sample
+
+This sample demonstrates how to use Firestore. At the moment, it only contains 
+code that show how to use the `@FirestoreQuery` property wrapper.
```
Comment (chFlorian) [inline Example/FirestoreSample/README.md:4]: 'code that show' -> 'code that shows'? I'm not sure about the correct grammar, but 'shows' sounds correct to me.

File: Example/FirestoreSample/FirestoreSample/Views/FavouriteFruitsView.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      }
+    }
+  }
+
+  func toggleFilter() {
+    showOnlyFavourites.toggle()
+    if showOnlyFavourites {
+      $fruits.predicates = [
+        .whereField("isFavourite", isEqualTo: true),
+        .where(field: "isFavourite", isEqualTo: true),
```
Comment (chFlorian) [inline Example/FirestoreSample/FirestoreSample/Views/FavouriteFruitsView.swift:54]: Why do we need both ways here? They are already described in the enum documentation. If you want to keep both ways I would suggest to also keep the third way (.isEqualTo(...)).

File: Firestore/Swift/Source/PropertyWrapper/QueryPredicate.swift (hunk 2 above)
Comment (peterfriese) [reply Firestore/Swift/Source/PropertyWrapper/QueryPredicate.swift:23]: This is mostly there to gather feedback (both from the community and from out internal API review process).

File: Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift (hunk 3 above)
Comment (peterfriese) [reply Firestore/Swift/Source/PropertyWrapper/FirestoreQuery.swift:63]: This is what Apple did with their `Configuration` type for `@FetchRequest` as well, so I was just following their example.

File: Example/FirestoreSample/README.md (hunk 4 above)
Comment (peterfriese) [reply Example/FirestoreSample/README.md:4]: Good catch! Updating this now.

File: Example/FirestoreSample/FirestoreSample/Views/FavouriteFruitsView.swift (hunk 5 above)
Comment (peterfriese) [reply Example/FirestoreSample/FirestoreSample/Views/FavouriteFruitsView.swift:54]: Good catch - I used this to prototype the API and forgot to remove one.

**Pull request merged**